repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DomainPreProcessor.java

```diff
@@ -93,7 +93,7 @@ private void processUpdateDomain(AtlasEntity entity, AtlasVertex vertex) throws
         String vertexQnName = vertex.getProperty(QUALIFIED_NAME, String.class);
 
         AtlasEntity storedDomain = entityRetriever.toAtlasEntity(vertex);
-        if(storedDomain.getRelationshipAttribute(PARENT_DOMAIN) == null){
+        if(vertex.getProperty(PARENT_DOMAIN, String.class) == null){
```

Comment:
Parent is a relationship to Domain hence vertex.getProperty() will not return it